src/transformers/modeling_tf_utils.py

```diff
@@ -2905,16 +2895,19 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         if safetensors_from_pt:
             from .modeling_tf_pytorch_utils import load_pytorch_state_dict_in_tf2_model
 
-            state_dict = safe_load_file(resolved_archive_file)
-            # Load from a PyTorch checkpoint
-            return load_pytorch_state_dict_in_tf2_model(
-                model,
-                state_dict,
-                allow_missing_keys=True,
-                output_loading_info=output_loading_info,
-                _prefix=load_weight_prefix,
-                ignore_mismatched_sizes=ignore_mismatched_sizes,
-            )
+            with safe_open(resolved_archive_file, framework="tf") as state_dict:
```

Comment:
I'm slightly horrified to see a handle be called a dict, but it's a clever hack to not rewrite too many things.

Reply:
I'll rename it here, and hopefully the docstring on the actual state dict loading function is clear enough!